src/sql/workbench/contrib/query/browser/gridPanel.ts

```diff
@@ -65,6 +65,8 @@ const ACTIONBAR_HEIGHT = 120;
 // this handles min size if rows is greater than the min grid visible rows
 const MIN_GRID_HEIGHT = (MIN_GRID_HEIGHT_ROWS * ROW_HEIGHT) + HEADER_HEIGHT + ESTIMATED_SCROLL_BAR_HEIGHT;
 
+const IsJsonRegex = /({.*?})/g;
```

Comment:
What about text values that contain { } in them but aren't valid JSON? I feel like it'd be better to try and parse it as JSON instead to test, although we might start having perf issues if we're doing that for large blobs of data...

Reply:
And even if you keep this logic, you're removing the check for the column data type being nvarchar which could cause problems too

Reply:
The purpose of this feature is to allow user to see the json content in a separate editor, as a result:
1. it doesn't need to be very accurate, using the regex is definitely much more performant than actually parsing it. so I think a minimum check is good enough.
2. nvarchar column type check is not required, we should allow user to view  the json data regardless of the type.

this is what I think. @Charles-Gagnon  @kburtram what do you think?

Reply:
Alright, guess this is enough of an improvement and there isn't any better solution I can come up with that this seems fine for now. Please add comments explaining this though so it's clear that we're just doing a "best low-cost effort" here and not trying to be 100% accurate.

Reply:
I'd also definitely suggest adding appropriate documentation to the STS code making it clear that it only checks the first row currently. 

And this might be a good time to look at the extension API and try to figure out why we even have the IDbColumn and IGridResult interfaces there. They only seem to be used internally currently.